### PR TITLE
Guaranty a device deletion event is published

### PR DESCRIPTION
**Pull request by koshelev**
- Web API endpoint publishes an event to kafka
- background job subscribes to events and deletes the device

### CODE REVIEW

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2017 ATS Advanced Telematic Systems GmbH
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala]: 2018

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+) {
+
+  def start(): Unit = {
+    val listener = system.actorOf(MessageListener.props[DeleteDeviceRequest](system.settings.config, handle, metrics),
+                                  "delete-device-handler")
+    listener ! Subscribe
+  }
+
+  private[this] def handle(message: DeleteDeviceRequest): Future[Done] =
+    db.run(DeviceRepository.delete(message.namespace, message.uuid).map(_ => Done))
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala]: I wonder about error handling for this mechanism. Would be good to at least have a debug log here for debugging.

Comment (koshelev) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala]: Flow will handle it - log and retry

Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala]: Ah cool. Where is that set up?

Comment (koshelev) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/daemon/DeleteDeviceHandler.scala]: Somewhere in `MessageListener`

**Pull request merged**